pkg/shed/index.go

```diff
@@ -276,13 +292,16 @@ func (f Index) Delete(keyFields Item) (err error) {
 
 // DeleteInBatch is the same as Delete just the operation
 // is performed on the batch instead on the database.
-func (f Index) DeleteInBatch(batch *leveldb.Batch, keyFields Item) (err error) {
+func (f Index) DeleteInBatch(batch *badger.Txn, keyFields Item) (err error) {
 	key, err := f.encodeKeyFunc(keyFields)
 	if err != nil {
 		f.logger.Debugf("keyfields encoding error in DeleteInBatch. Error: %s", err.Error())
 		return err
 	}
-	batch.Delete(key)
+	err = batch.Delete(key)
+	if err != nil {
+		f.logger.Debugf("could not delete items in batch. Error: %s", err.Error())
```

Comment:
Why the error is not passed to the caller?

This may result in committing an incomplete batch.

Reply:
Do not log error and return it.